# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerError.cpp
# Commit: ffc69aa8f2c3
# Full Hash: ffc69aa8f2c3bcdc3cc4f2516a618c0430e67ddf
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P2 WorkerParentThreadRunnable for the runnables dispatched to worker's parent thread. r=dom-worker-reviewers,asuth
#   
#   In this patch, WorkerParentThreadRunnable is extracted from WorkerThreadRunnable for runnable on the parent thread.
#   
#   WorkerParentControlRunnable and WorkerParentDebuggeeRunnable are also created for control runnable and debuggee runnable on the parent thread.
# ==============================================================================

diff -r 379cd5044768 -r ffc69aa8f2c3 dom/workers/WorkerError.cpp
--- a/dom/workers/WorkerError.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/workers/WorkerError.cpp	Fri Apr 19 09:41:57 2024 +0000
@@ -62,13 +62,13 @@
 
 namespace {
 
-class ReportErrorRunnable final : public WorkerDebuggeeRunnable {
+class ReportErrorRunnable final : public WorkerParentDebuggeeRunnable {
   UniquePtr<WorkerErrorReport> mReport;
 
  public:
   ReportErrorRunnable(WorkerPrivate* aWorkerPrivate,
                       UniquePtr<WorkerErrorReport> aReport)
-      : WorkerDebuggeeRunnable(aWorkerPrivate, "ReportErrorRunnable"),
+      : WorkerParentDebuggeeRunnable(aWorkerPrivate, "ReportErrorRunnable"),
         mReport(std::move(aReport)) {}
 
  private:
@@ -138,7 +138,7 @@
   }
 };
 
-class ReportGenericErrorRunnable final : public WorkerDebuggeeRunnable {
+class ReportGenericErrorRunnable final : public WorkerParentDebuggeeRunnable {
  public:
   static void CreateAndDispatch(WorkerPrivate* aWorkerPrivate) {
     MOZ_ASSERT(aWorkerPrivate);
@@ -151,7 +151,8 @@
 
  private:
   explicit ReportGenericErrorRunnable(WorkerPrivate* aWorkerPrivate)
-      : WorkerDebuggeeRunnable(aWorkerPrivate, "ReportGenericErrorRunnable") {
+      : WorkerParentDebuggeeRunnable(aWorkerPrivate,
+                                     "ReportGenericErrorRunnable") {
     aWorkerPrivate->AssertIsOnWorkerThread();
   }
 